test/e2e/framework/pod/resource.go

```diff
@@ -259,28 +259,30 @@ func DumpAllPodInfoForNamespace(ctx context.Context, c clientset.Interface, name
 	logPodLogs(ctx, c, namespace, pods.Items, reportDir)
 }
 
-// FilterNonRestartablePods filters out pods that will never get recreated if
+// FilterNonRecreatablePods filters out pods that will never get recreated if
 // deleted after termination.
-func FilterNonRestartablePods(pods []*v1.Pod) []*v1.Pod {
+func FilterNonRecreatablePods(pods []*v1.Pod) []*v1.Pod {
 	var results []*v1.Pod
 	for _, p := range pods {
-		if isNotRestartAlwaysMirrorPod(p) {
-			// Mirror pods with restart policy == Never will not get
-			// recreated if they are deleted after the pods have
-			// terminated. For now, we discount such pods.
-			// https://github.com/kubernetes/kubernetes/issues/34003
+		if isPodNotRecreatedIfDeleted(p) {
 			continue
 		}
 		results = append(results, p)
 	}
 	return results
 }
 
-func isNotRestartAlwaysMirrorPod(p *v1.Pod) bool {
-	// Check if the pod is a mirror pod
-	if _, ok := p.Annotations[v1.MirrorPodAnnotationKey]; !ok {
-		return false
+func isPodNotRecreatedIfDeleted(p *v1.Pod) bool {
```

Comment:
nit: avoid "not" in function names. Better to invert the logic and call this `isPodRecreatedIfDeleted`

Reply:
yep , fixed